src/stdune/path.mli

```diff
@@ -24,16 +43,18 @@ module Source : sig
     val relative : ?error_loc:Loc0.t -> t -> string list -> t
   end
 
+  val of_relative : Relative.t -> t
   val of_local : Local.t -> t
   val relative : ?error_loc:Loc0.t -> t -> string -> t
-  val split_first_component : t -> (string * Local.t) option
+  val split_first_component : t -> (string * t) option
```

Comment:
I don't think this change should be made: if you split first component from source path, you generally no longer have a source path.

Reply:
Oh yeah, I will undo this.